File: src/utils/logger.ts
enum Colour {
    Black   = '\u001b[30m',
    Red     = '\u001b[31m',
    Green   = '\u001b[32m',
    Yellow  = '\u001b[33m',
    Blue    = '\u001b[34m',
    Magenta = '\u001b[35m',
    Cyan    = '\u001b[36m',
    White   = '\u001b[37m',

    Default = '\u001b[0m',
}

export class Logger {
    static log(log: string) {
        console.log(Colour.White  + `LOG   : ${log}` + Colour.Default);
    }

    static warning(log: string) {
        console.log(Colour.Yellow + `WARN  : ${log}` + Colour.Default);
    }
    
    static error(log: string) {
        console.log(Colour.Red    + `ERR   : ${log}` + Colour.Default);
    }

    /** Use this when receiving a message from Discord */
    static discord(log: string) {
        console.log(Colour.Blue   + `DISCO : ${log}` + Colour.Default);
    }
}

File: Tools/Logger.cs
namespace MoeBot.Tools {
    internal static class Logger {
        private static void Log(string message, ConsoleColor colour) {
            Console.ForegroundColor = colour;
            Console.WriteLine(message);
            Console.ResetColor();
        }

        internal static void Log(string message) {
            DateTime dateTime = DateTime.Now;
            Log($"{dateTime.ToString("HH:mm:ss")} Log         {message}", ConsoleColor.White);
        }

        internal static void DiscordCommand(string message) {
            DateTime dateTime = DateTime.Now;
            Log($"{dateTime.ToString("HH:mm:ss")} Command     {message}", ConsoleColor.Blue);
        }

        internal static void DiscordMessage(string message) {
            DateTime dateTime = DateTime.Now;
            Log($"{dateTime.ToString("HH:mm:ss")} Message     {message}", ConsoleColor.Green);    
        }

        internal static void Error(string message) {
            DateTime dateTime = DateTime.Now;
            Log($"{dateTime.ToString("HH:mm:ss")} Error       {message}", ConsoleColor.Red);
        }

        internal static void Warning(string message) {
            DateTime dateTime = DateTime.Now;
            Log($"{dateTime.ToString("HH:mm:ss")} Warning     {message}", ConsoleColor.Yellow);
        }
    }
}